Lambdas/GHHooks/SecretsRetriever.swift

```diff
@@ -23,6 +41,30 @@ struct SecretsRetriever {
         guard let secret = secret.secretString else {
             throw Errors.secretNotFound(arn: arn)
         }
-        return secret
+        return Secret(secret)
+    }
+
+    private func withLock<T>(block: () async throws -> T) async rethrows -> T {
+        if isLoading {
+            await withCheckedContinuation { continuation in
+                loadWaiters.append(continuation)
+            }
+        }
+
+        isLoading = true
+        defer { isLoading = false }
+
+        let value = try await block()
+
+        for waiter in self.loadWaiters {
+            waiter.resume()
+        }
+        self.loadWaiters.removeAll()
```

Comment:
```suggestion
    isLoading = false
    self.loadWaiters.popLast()?.resume()
```
If you resume multiple waiters, you'll just wake them all up just to loop all but one of them back into another round of continuation awaits. By waking up only one, you get rid of a bunch of completely pointless contention on the Concurrency runtime's internal locks. It also avoids the possible race condition where two or more waiters get past the `isLoading` check before it gets set to `true` again.

But you do need to de-flag `isLoading` first, otherwise you'll very likely deadlock (`isLoading` will still be true, so it'll loop again, but there'll be no active loads to trigger another resume). You should remove the `defer` above too, un-flagging it twice can cause a different kind of TOCTOU race.